Sprint 17

Depechist opened the pull request.

Depechist committed:
Статистика + Темная тема

В работе.

- Создан экран статистики с логикой. Баг: экран всегда показывает коллекцию, будет исправлено.
- Добавлена темная тема

Depechist committed:
Аналитика

- Добавлена Яндекс.Метрика и События на следующие эвенты:

- Открытие
- Закрытие
- Тап на добавление трекера
- Тап на кнопку о выполнении трекера

Depechist committed:
Статистика + Темная тема. Исправления

- Исправлен экран статистики. Теперь он корректно реагирует на количество трекеров или их отсутствие
- Статистика теперь корректно отображает количество выполненных трекеров
- Исправлен баг с цветом подложки таблицы на экране Статистики
- Таблицу на экране Статистики теперь нельзя скроллить
- Исправлен баг с некорректным отображением плейсхолдера на экране Статистики
- Исправлено поведение экрана Онбординга. Теперь элементы не меняют свой цвет, если включена темная тема - согласно дизайну в Фигме.

Depechist committed:
Исправления

- Исправлена "прозрачность" tabBar (исправлен нижний констрейнт у коллекции), а также добавлена полоска сверху согласно дизайну в Фигме
- В TrackersViewController добавлены плейсхолдеры для пустой выдачи в поиске, необходимо дописать логику под корректное отображение

Depechist committed:
Скриншот-тест

- Добавлен скриншот-тест, который успешно пройден

Depechist committed:
Контекстное меню + другое

- Добавлено контекстное меню с функционалом Закрепа, Редактирования и Удаления трекера
- Добавлена аналитика на События по действиям в контекстном меню
- На главный экран добавлена кнопка Фильтры
- Добавлена аналитика на тапы по кнопке Фильтры
- Дописана логика отображения плейсхолдера emptySearch в случае пустой коллекции при поиске Трекеров

Depechist committed:
Локализация

- Добавлены локализации: русский язык, английский язык

amardanov commented on Tracker/Helpers/Analytics.swift:
```diff
@@ -0,0 +1,48 @@
+import Foundation
+import YandexMobileMetrica
+
+final class Analytics {
```
Принято называть такие классы с `Service` в конце. То есть `AnalyticsService`

Depechist committed:
Правки

- Файл со скриншот-тестами возвращен на место. Тесты успешно пройдены.
- Скролл коллекции на главном экране скорректирован, чтобы нижние ячейки не попадали под кнопку Фильтры.

Depechist committed:
Фильтры + другое

- События open и close Аналитики изменены на открытие и закрытие экрана с трекерами
- Написана логика работы кнопки Фильтры
